lib/core/test/unit/Cardano/Wallet/DB/Store/Transactions/StoreSpec.hs

```diff
@@ -14,23 +19,172 @@ import Cardano.Wallet.DB.Fixtures
 import Cardano.Wallet.DB.Sqlite.Types
     ( TxId (TxId) )
 import Cardano.Wallet.DB.Store.Transactions.Model
-    ( DeltaTxHistory (..), TxHistoryF (..), mkTxHistory )
+    ( DeltaTxHistory (..)
+    , TxHistory
+    , TxHistoryF (..)
+    , WithTxOut (..)
+    , collateralIns
+    , decorateWithTxOuts
+    , ins
+    , mkTxCollateral
+    , mkTxHistory
+    , mkTxIn
+    , mkTxOut
+    , undecorateFromTxOuts
+    )
 import Cardano.Wallet.DB.Store.Transactions.Store
     ( mkStoreTransactions )
+import Cardano.Wallet.Primitive.Types.Tx
+    ( Tx (..) )
+import Control.Monad.State.Strict
+    ( MonadState (get), evalState, modify )
+import Data.Functor
+    ( (<&>) )
+import Data.Generics.Internal.VL
+    ( set )
+import List.Transformer
+    ( select )
 import Test.DBVar
     ( GenDelta, prop_StoreUpdates )
 import Test.Hspec
-    ( Spec, around, describe, it )
+    ( Spec, around, describe, it, shouldBe )
 import Test.QuickCheck
-    ( arbitrary, elements, frequency, property )
+    ( Gen
+    , Property
+    , arbitrary
+    , elements
+    , forAll
+    , frequency
+    , property
+    , shuffle
+    )
 
+import qualified Cardano.Wallet.Primitive.Types.Coin as W
+import qualified Cardano.Wallet.Primitive.Types.Tx as W
+import qualified Control.Foldl as L
 import qualified Data.Map.Strict as Map
+import qualified List.Transformer as LT
 
 spec :: Spec
-spec = around withDBInMemory $ do
-    describe "Transactions store" $ do
-        it "respects store laws"
-            $ property . prop_StoreLaws
+spec = do
+    around withDBInMemory $ do
+        describe "Transactions store" $ do
+            it "respects store laws" $
+                property . prop_StoreLaws
+    describe "TxOut decoration" $ do
+        it "respects order and content of transactions" $
+            property prop_DecorateIsInvertible
+        it
+            "reports a transaction where inputs point \
+            \to all other transactions output"
+            $ property prop_DecorateLinksTxInToTxOuts
+        it
+            "reports a transaction where collateral inputs point \
+            \to all other transactions output"
+            $ property prop_DecorateLinksTxCollateralsToTxOuts
+
+prop_DecorateIsInvertible :: [Tx] -> Bool
+prop_DecorateIsInvertible transactions =
+    let txh = mkTxHistory transactions
+     in undecorateFromTxOuts (decorateWithTxOuts txh) == txh
```

Comment:
```suggestion
    in  undecorateFromTxOuts (decorateWithTxOuts txh) == txh
```
(Style: Seems to be the best of both worlds. 4 spaces indentation, _both_ before and after `in`. 🤔 )